### PR TITLE
Add groovy library for smoke test workflow

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Add groovy library for smoke tests

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/5190

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: vars/runSmokeTestScript.groovy [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
```
Comment (peterzhuamazon) [inline vars/runSmokeTestScript.groovy:8]: Could you add comments to list all the args

File: build.gradle [hunk 2]
```diff
@@ -128,7 +128,7 @@ jacocoTestReport {
     }
 }
 
-String version = '8.0.0'
+String version = '8.0.1'
```
Comment (peterzhuamazon) [inline build.gradle:131]: Should be 8.1.0 as this is adding a new lib.

File: vars/runSmokeTestScript.groovy (hunk 1 above)
Comment (zelinh) [reply vars/runSmokeTestScript.groovy:8]: Added. Thanks.

File: build.gradle (hunk 2 above)
Comment (zelinh) [reply build.gradle]: Updated. Thanks.

File: vars/runSmokeTestScript.groovy [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**@
+ * Contains logic to execute the smoke test workflow
+ *
+ * @param args A map of the following parameters
+ * @param args.testManifest <required> Test manifest file location
+ * @param args.buildManifest <required> Build manifest file location
+ * @param args.jobName <optional> Job name that triggered the workflow. 'distribution-build-opensearh' by default.
```
Comment (peterzhuamazon) [inline vars/runSmokeTestScript.groovy:15]: Nit: make optional param at the end of the list.

**Review by peterzhuamazon — APPROVED** [review]

File: vars/runSmokeTestScript.groovy (hunk 3 above)
Comment (zelinh) [reply vars/runSmokeTestScript.groovy]: Updated. Thanks!

**Pull request merged**